### PR TITLE
trivial: dell-dock: check for valid dock type on open

### PR DESCRIPTION
**Pull request by superm1**
Only add instance ID if it actually probes properly.

Otherwise this makes an invalid assumption that the device is a WD19
EC just because it had the correct hub in front.

Instead check the first time it's opened that the correct device
is identified (`EXPECTED_DOCK_TYPE`)

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/dell-dock/fu-dell-dock-i2c-ec.c [hunk 1]
```diff
@@ -312,12 +312,24 @@ fu_dell_dock_is_valid_dock (FuDevice *device, GError **error)
...
-		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_NOT_FOUND,
+	if (result == NULL) {
+		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
 				     "No valid dock was found");
 		return FALSE;
 	}
-	return TRUE;
+
+	/* this will trigger setting up all the quirks */
+	if (*result == EXPECTED_DOCK_TYPE) {
```
Comment (hughsie) [inline plugins/dell-dock/fu-dell-dock-i2c-ec.c:322]: are we sure *result is valid? I'd feel a lot more content getting the size from `g_bytes_get_data` and also checking the `sz >= 1`

**Comment by hughsie** [comment]
One niggle then LGTM, thanks.

File: plugins/dell-dock/fu-dell-dock-i2c-ec.c (hunk 1 above)
Comment (superm1) [reply plugins/dell-dock/fu-dell-dock-i2c-ec.c:322]: Right before checked that result wasn't NULL

Comment (superm1) [reply plugins/dell-dock/fu-dell-dock-i2c-ec.c:322]: But I guess that's a cleaner style way to check via size. I'll adjust.

Comment (superm1) [reply plugins/dell-dock/fu-dell-dock-i2c-ec.c]: fixed

**Review by hughsie — APPROVED** [review]

**Pull request merged**